### PR TITLE
add data to dv

### LINKED ISSUE #4: dataverse uploader

### PR DESCRIPTION
**Pull request by atrisovic**
Closes #4

### CODE REVIEW

File: spacebench/api/dataverse.py [hunk 1]
```diff
@@ -62,6 +63,22 @@ def __get_fileids_from_filenames(self):
             if filename in self.files.keys():
                 self.files[filename] = file["dataFile"]["id"]
 
+    def list_data_files(self, include_fileid=False):
+        """ Lists data files from Dataverse. """
+
+        files_list = self.dataset.json()[
+            'data']['latestVersion']['files']
```
Comment (Naeemkh) [inline spacebench/api/dataverse.py:70]: To ensure consistency, please choose either single or double quotes and stick with your choice. I recommend using double quotes.

File: spacebench/api/dataverse.py [hunk 2]
```diff
@@ -101,10 +117,63 @@ def download_data(self, predictor, type):
                 print("Downloaded: file name {}, id {}".format(
                     filename, self.files[filename]))
 
+    def publish_dataset(self, token):
```
Comment (Naeemkh) [inline spacebench/api/dataverse.py:120]: Please add documentation.

File: spacebench/api/dataverse.py [hunk 3]
```diff
@@ -101,10 +117,63 @@ def download_data(self, predictor, type):
...
+        """ Publish new dataset. """
+        api = NativeApi(self.base_url, token)
+        resp = api.publish_dataset(self.pid, release_type="major")
+        if resp.json()["status"] == "OK":
+           print("Dataset published.")
 
-    def upload_data(self):
+    def upload_data(self, file_path, description, token):
```
Comment (Naeemkh) [inline spacebench/api/dataverse.py:127]: Please add argument docs.

File: spacebench/api/cli.py [hunk 4]
```diff
@@ -1,59 +1,137 @@
...
+            f"Replacing datafile with ID {args.file_id} with file: {args.file}")
+        self.dvapi.replace(args.file, args.file_id, args.token)
+
+    def publish_uploaded_dataset(self, args):
+        self.dvapi.publish_dataset(args.token)
+
+    def list_data_files(self, args):
+        """ Lists all existing data files. """
+        logging.info("Existing data files:")
+        print(self.dvapi.list_data_files(args.include_fileid))
```
Comment (Naeemkh) [inline spacebench/api/cli.py:105]: To ensure proper rendering, please include all printings in the logging message.

File: spacebench/api/cli.py [hunk 5]
```diff
@@ -1,59 +1,137 @@
...
 
+    def download_and_generate_data(self, args):
+        """ Downloads data core and generates samples. """
         target_dir = Path(args.output_path)
```
Comment (Naeemkh) [inline spacebench/api/cli.py:109]: Include this block in a `try-catch` block.

**Review by Naeemkh — COMMENTED** [review]
@atrisovic, thanks for the PR. I put some minor comments. Please address them and push your changes.

File: spacebench/api/cli.py [hunk 6]
```diff
@@ -1,59 +1,137 @@
 """ Command-line client """
 
 import os
+import logging
 import argparse
+from pathlib import Path
 from spacebench.api.dataverse import DataverseAPI
```
[the remark this answers is not in the document]
Comment (atrisovic) [reply spacebench/api/cli.py:7]: You mean a new line between 5 and 6?

**Comment by atrisovic** [comment]
I incorporated the changes and created a new commit.

**Review by Naeemkh — APPROVED** [review]

**Pull request merged**